### ISSUE TITLE
How can I add link buttons  in Top App Bar?

### ISSUE THREAD

**Issue by shirleyhsliu81** [material-components/material-components-web-components#1945]
Hi There,

I am trying to use Top App Bar. Please see attached screenshot. In my design, I need to add some links (buttons) in the bar on the left of the icon buttons. When a link / a module is selected, the link will be underlined.

I believe this is a quite common design for banner. Could you please advise how I can achieve this with Top App Bar? Many thanks.

Cheers,
Shirley

**Comment by taylor-vann** [comment]
Hi there @shirleyhsliu81 

Unfortunately I don't see any screenshots attached to your comment. If you could provide us with sample code as well that would be awesome.

A clarifying question would be: do you need to use top-app-bar for your requirements?
I feel Top App Bar has very specific use cases involved and the "form" of the Top App Bar is tied to its content.

However making a component more tailored to your own use case (a banner) should be very doable with [LitElement](https://lit-element.polymer-project.org/).

**Comment by shirleyhsliu81** [comment]
<img width="1165" alt="Top App Bar link buttons" src="https://user-images.githubusercontent.com/58677163/99009617-6a17c280-259c-11eb-9b6d-516b2f5aa41a.png">

Hi @taylor-vann,

Thanks for your reply.

Please check the screenshot above and see if there is any easy way to include the link button in the top app bar. I think it is quite common in material design.

Many Thanks,
Shirley

**Comment by taylor-vann** [comment]
Hey @shirleyhsliu81

Thanks for the screenshot, it's nice to see what you're building!

TLDR; you might want to look into [mwc-tab-bar](https://github.com/material-components/material-components-web-components/tree/master/packages/tab-bar) or [LitElement](https://lit-element.polymer-project.org) to roll your own App Bar.

You're correct, those link buttons in your screenshot have a very material feel to them. The only problem is App Bar was designed with very specific cases in mind.  The ability to extend App Bar in terms of internal content is highly curated.

Your case is visually similar but different enough from our implementation that it might actually be faster to build an App Bar.
